src/utils/cspUtils.js

```diff
@@ -0,0 +1,133 @@
+  const schemes = ['http:', 'https:'];
+  const dataschemes = ['data:', 'mediastream:', 'blob:', 'filesystem:'];
+
+  const schemesSatisfied = schemes.some(scheme => (_stringContainsValue(policy, scheme) && _.startsWith(elemSrc, scheme)) );
+  const dataschemesSatisfied = dataschemes.some(datascheme => (_stringContainsValue(policy, datascheme) && _.startsWith(elemSrc, datascheme)) );
+  return schemesSatisfied || dataschemesSatisfied;
+};
+
+function  _checkSelfPolicy (elemSrc, policy) {
+  return ( isLinkInternal(elemSrc) && _stringContainsValue(policy, "'self'"));
+};
+
+/* Helper function to check if elemAttr satisfies CSP source specifications */
+function _elemAttrSatisfiesPolicies (elemAttr, policy) {
+  if (_stringContainsValue(policy, "'none'")) return false; 
+  
+  const selfSatisfied = _checkSelfPolicy(elemAttr, policy);
+  const schemasourceSatisfied = _checkSchemasourcePolicy(elemAttr, policy);
+  const hostsourceSatisfied = _checkHostsourcePolicy(elemAttr, policy);
+  return selfSatisfied || schemasourceSatisfied || hostsourceSatisfied;
+};
+
+/* Helper function to get resource policy from csp for given policyType*/
+function _getResourcePolicy(csp, policyType) {
+  const resourcePolicyMapping = { 
+    'frame-src': ['frame', 'iframe'],
+    'img-src': ['img'],
+    'media-src': ['audio', 'video', 'track'],
+    'object-src': ['object', 'embed', 'applet'],
+    'script-src-elem': ['script'],
+  };
+  const cspPolicy = new Policy(csp);
+
+  let resourcePolicy
+  if (policyType === 'frame-src') { // from http://csplite.com/csp/test121/, fallback chain: frame-src -> child-src -> default-src
+    resourcePolicy = cspPolicy.get(policyType) ||  cspPolicy.get('child-src') || cspPolicy.get('default-src')
+  } else if (policyType === 'script-src-elem') { // fallback chain: script-src-elem -> script-src -> default-src
+    resourcePolicy = cspPolicy.get(policyType) ||  cspPolicy.get('script-src') || cspPolicy.get('default-src')
+  } else {
+    resourcePolicy = cspPolicy.get(policyType) === '' ? cspPolicy.get('default-src') : cspPolicy.get(policyType);
+  }
+  const resourcePolicyElems = resourcePolicyMapping[policyType];
+  return { resourcePolicy, resourcePolicyElems } 
+}
+
+/* Helper function to check if resourcePolicyElems for a specific policyType satisfies CSP source specifications */
+function _checkResourcePolicyElems(resourcePolicyElems, policy, $) {
+  let policyViolation = false;
+  resourcePolicyElems.forEach(elemType => { 
+    $(elemType).each((i, elem) => {
+      const checkAttr = elemType === 'object' ? 'data' : 'src' // exception for object html: <object data='abc.html'></object>
+      if (!_elemAttrSatisfiesPolicies($(elem).attr(checkAttr), policy)) {
+        $(elem).replaceWith(`<span style="color:#c91508"> Intended &lt${elemType}&gt content violates Content Security Policy and therefore could not be displayed. Isomer does not support display of any forbidden resources. </span>`);
```

Comment:
Nit: What is the rationale for using `<span>` instead of `<p>`? Minor issue, but if there are subsequent violations, there is no line break between the warning text if we use `span`.

<img width="1669" alt="Screenshot 2020-11-10 at 6 25 54 AM" src="https://user-images.githubusercontent.com/31984694/98603967-eb155680-231d-11eb-9b2e-7ba30cb42706.png">

Reply:
I was following the format for error messages in form fields, for example [here](https://github.com/isomerpages/isomercms-frontend/blob/5daede304bb09c8752748168459afc7eae564d71/src/components/FormFieldMedia.jsx#L126), but I can replace it with `<p>` too, what do you think?

Reply:
600897b

Reply:
Yup line break works